test/dynamo/test_aot_autograd.py

```diff
@@ -281,6 +285,56 @@ def capturing_fn(gm, inputs):
         aot_fn(x, y)
         self.assertTrue(is_safe[0])
 
+    @patch("functorch._src.config.debug_assert", True)
+    def test_requires_grad_fake_via_dynamo_recompiles(self):
+        shape_env = ShapeEnv()
+        fake_mode = FakeTensorMode(shape_env=shape_env)
+
+        class F(torch.nn.Module):
+            def forward(self, x, y):
+                return (x + y,)
+
+        x = torch.randn(3, 3, requires_grad=True)
+        y = torch.randn(3, 3, requires_grad=True)
+        z = torch.randn(3, 3, requires_grad=False)
+
+        fake_x = fake_mode.from_tensor(x)
+        fake_y = fake_mode.from_tensor(y)
+        fake_z = fake_mode.from_tensor(z)
```

Comment:
It's not clear to me why in this test we need to manually fakeify the tensors here; we're running dynamo and dynamo should be managing all of this

Reply:
Because it helps me test that it works correctly, as it simulates something dynamo does without the dynamo parts above.